.codeclimate.yml

```diff
@@ -0,0 +1,23 @@
+exclude_patterns:
+- "cypress/**"
+- "**/*.test.*"
+- "**/*.stories.*"
+- "**/config/**"
+- "**/fixture*"
+- "scripts/lighthouseBudget.js"
+- "scripts/lighthouseConfig.js"
+- "src/index.js"
+- "src/server/index.jsx"
+- "src/app/components/Metadata/index.jsx"
```

Comment:
Very curious of the decision to exclude so many of these files from our code climate coverage. These are v important files in our application and it worries me were just disabling code climate on them, im assuming because they have 1 or 2 issues on them? 
Whats wrong with these remaining as issues in our code? 
Will this mean any more issues we add to these files, that code climate would spot, would now be missed?
If we must, can we disable specific rules for files, rather than a blanket exclude?

Reply:
Agree - I was surprised by how many there are too, and some of the files are pretty core

These were just the files that Code Climate reported to have "new issues" as a result of just adding a `.codeclimate.yml file` - 250 new issues blocking merge

<img width="623" alt="Screen Shot 2019-07-05 at 16 09 29" src="https://user-images.githubusercontent.com/8601460/60731466-84b94080-9f3f-11e9-9223-53fdc8806fcf.png">

All of the issues were actually pre-existing, so I just assumed they were being suppressed by the web-based config, so I added those files back to the exclusion list

However, @benjaminhobbs posted the exception list from the web config, and they did not include many of these files.  I think that leaves 2 possibilities:

- These issues were being suppressed in some other way
- The issues were "grandfathered in" in some way (admin merges?  issues pre-date the introduction of code climate?)

Either way, I'm not advocating we just ignore these files - this is just a placeholder PR until I (or someone else) picks up the issue and investigates it properly :)

Reply:
>These issues were being suppressed in some other way
This is correct, we should ignore the 250 issues to fix as a problem for this PR and only post the excludes we already have (copy below).

Rather than the manual overrides we had before (in the CC interface) we can then manage them by appending to the exclude file when necessary.

This current formal exclude list is:
config/
db/
dist/
features/
**/node_modules/
script/
**/spec/
**/test/
**/tests/
Tests/
**/vendor/
**/*_test.go
**/*.d.ts
cypress/{integration,support}/*.js
src/**/*.test.{jsx,js}
src/app/lib/config/**

Reply:
Because of the 250 warnings, this will need admin to merge